### DISCUSSION TITLE
Default configuration from Grails create-app

### DISCUSSION THREAD

**Discussion by jamesfredley** [apache/grails-core#13829] [Polls]
Configuration in generated apps includes the following locations:

- `application.yml`
- `build.gradle`
- `logback-spring.xml`

**Comment by jdaugherty** [comment]
I'd vote for defaulting to minimal, but adding a page similar to Micronaut's documents for just configuration so that we can easily see the defaults.  I'd also suggest we add a command on the CLI to show those defaults.

**Comment by matrei** [comment]
> but adding a page similar to Micronaut's documents for just configuration

@jdaugherty Is it this page you are referencing https://docs.micronaut.io/latest/guide/configurationreference.html?
This is a good idea. (It does however seem like Micronaut's page does not include the default values for most configuration properties.)

**Reply by jdaugherty** [reply]
I was thinking something like that, yes.  Assume we could generate it as part of the build process.

**Reply by jamesfredley** [reply]
https://docs.spring.io/spring-boot/appendix/application-properties/index.html is also a good example

**Comment by gsartori** [comment]
@jdaugherty I like the idea of having a CLI command to display the values of the whole application configuration regardless whether those values are the default ones or the configured ones in `application.yml`. I don't think it is going to be something easy to implement though, but let's deram, it would be useful. And it could work offline. That could become the way to document configuration properties both in Grails and in Grails plugins.
